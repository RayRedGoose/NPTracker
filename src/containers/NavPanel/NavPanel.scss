@import "../../variables.scss";

.nav-panel {
  border-bottom: 2px solid $darken-smallest;
  margin: 20px;
  padding-bottom: 10px;
  padding-right: 80px;
  width: fit-content;
  height: fit-content;
  .nav-tab {
    color: $light-color;
    font-family: 'Salsa';
    font-size: 18px;
    letter-spacing: 0.05em;
    text-decoration: none;
    margin: 0 10px;
    padding: 20px 10px 10px;
    &:hover {
      color: $accent-color-primary-dark;
      border-bottom: 2px solid $darken-small;
    }
    &.tab-active {
      color: $accent-color-primary-dark;
      border-bottom: 2px solid $accent-color-primary-dark;
    }
  }
}
