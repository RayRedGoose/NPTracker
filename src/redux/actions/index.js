export const addProcessName = name => ({
  type: 'ADD_PROCESS_NAME',
  name
})

export const addProcessStep = step => ({
  type: 'ADD_PROCESS_STEP',
  step
})

export const increaseStep = () => ({
  type: 'INCREASE_STEP'
})

export const addUserInfo = info => ({
  type: 'ADD_USER_INFO',
  info
})

export const addParks = parks => ({
  type: 'ADD_PARKS',
  parks
})

export const changeActiveTab = id => ({
  type: 'CHANGE_ACTIVE_TAB',
  id
})
