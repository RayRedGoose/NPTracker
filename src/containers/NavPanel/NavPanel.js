import './NavPanel.scss'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { NavLink } from 'react-router-dom'
import { changeActiveTab } from 'redux/actions'

export class NavPanel extends Component {
  toggleTab = (event) => {
    this.props.changeActiveTab(+event.target.id)
  }

  createTabs = () => {
    const tabNames = ['dashboard', 'parks', 'wish list', 'visited']

    return tabNames.map((tab, index) => {
      const tabClass = (this.props.activeTab === (index + 1)) ? 'nav-tab tab-active' : 'nav-tab'
      const link = tab.replace(/ /g, '-')
      return <NavLink
        to={`/${link}`}
        key={link}
        id={index+1}
        className={tabClass}
        onClick={this.toggleTab}>{ tab }</NavLink>
    })
  }

  render() {
    return (
      <nav className="nav-panel">
        { this.createTabs() }
      </nav>
    )
  }
}

export const mapStateToProps = ({ activeTab }) => ({
  activeTab
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    changeActiveTab
  }, dispatch)
)

NavPanel.propTypes = {
  activeTab: PropTypes.number.isRequired,
  changeActiveTab: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(NavPanel)
