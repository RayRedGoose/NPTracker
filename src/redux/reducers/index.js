import { combineReducers } from 'redux'
import { process } from './singleReducers/process'
import { user } from './singleReducers/user'

const rootReducer = combineReducers({
  process,
  user
})

export default rootReducer
