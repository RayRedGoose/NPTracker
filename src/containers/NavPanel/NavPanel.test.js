import React from 'react'
import { shallow } from 'enzyme'
import { NavPanel, mapStateToProps, mapDispatchToProps } from './NavPanel'
import { changeActiveTab } from 'redux/actions'

describe("NavPanel", () => {
  let panel, instance
  const changeActiveTab = jest.fn()

  beforeEach(() => {
    panel = shallow(
      <NavPanel
        activeTab={1}
        changeActiveTab={changeActiveTab} />
    )

    instance = panel.instance()
  })

  it("should match snapshot with all data passed in correctly", () => {
    expect(panel).toMatchSnapshot()
  })

  it("should call createTabs method after rendering", () => {
    const spy = jest.spyOn(instance, 'createTabs')
      .mockImplementation(jest.fn())
    instance.forceUpdate()

    expect(spy).toHaveBeenCalled()
  })

  it("should call changeActiveTab with target id when toggleTab is called", () => {
    const mockEvent = {target: {id: 2}}
    instance.toggleTab(mockEvent)
    expect(changeActiveTab).toHaveBeenCalledWith(2)
  })
})

describe("mapStateToProps", () => {
  it("should return the object with correct keys", () => {
    const mockStore = {
      activeTab: 1,
      user: {}
    }

    const expected = {
      activeTab: 1
    }

    const result = mapStateToProps(mockStore)

    expect(result).toEqual(expected)
  })
})

describe("mapDispatchToProps", () => {
  it("should call dispatch with changeActiveTab when changeActiveTab is called", () => {
    const mockDispatch = jest.fn()
    const mockProps = mapDispatchToProps(mockDispatch)
    const actionToDispatch = changeActiveTab(2)

    mockProps.changeActiveTab(2)

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})
