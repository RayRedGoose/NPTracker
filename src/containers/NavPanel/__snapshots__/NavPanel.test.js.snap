// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavPanel should match snapshot with all data passed in correctly 1`] = `
<nav
  className="nav-panel"
>
  <NavLink
    className="nav-tab tab-active"
    id={1}
    key="parks"
    onClick={[Function]}
    to="/parks"
  >
    parks
  </NavLink>
  <NavLink
    className="nav-tab"
    id={2}
    key="wish-list"
    onClick={[Function]}
    to="/wish-list"
  >
    wish list
  </NavLink>
</nav>
`;
